### ISSUE TITLE
Matplotlib throws the warning QApplication: invalid style override passed, ignoring it

### ISSUE THREAD

**Issue by aweinstein** [matplotlib/matplotlib#11775]
### Bug report

**Bug summary**

Using Matplotlib throws the warning `QApplication: invalid style override passed, ignoring it`. Same behavior as described [here](https://stackoverflow.com/questions/49799080/where-is-this-warning-being-raised-qapplication-invalid-style-override-passed).

It only happens the first time I call the plot command. Subsequent calls do not throw the warning.

**Code for reproduction**

```python
import matplotlib.pyplot as plt
plt.plot([1, 2])
```

**Actual outcome**

I get the warning `QApplication: invalid style override passed, ignoring it` plus the correct figure.

**Expected outcome**

Get the plot without warning.

**Matplotlib version**
  * Operating system: Linux Mint 18.1
  * Matplotlib version: 2.2.2
  * Matplotlib backend (`print(matplotlib.get_backend())`): Qt5Agg
  * Python version: 3.6.5
  * Other libraries: pyqt 5.9.2, qt 5.9.5

Everything installed from standard Anaconda distribution.

**Comment by tacaswell** [comment]
If you start up a black python terminal and do

```python
from PyQt5 import QtWidgets
qApp = QtWidgets.QApplication([str(" ")])
```

do you get the same warning?

**Comment by aweinstein** [comment]
@tacaswell Yes.

**Comment by tacaswell** [comment]
@aweinstein I am going to close this as not a Matplotlib bug (because you can reproduce it using only PyQt).

I am not sure that this is a bug, but strongly suspect that this is an interaction between your OS Qt's version and the Qt version and some global configuration.  If I understand correctly, Qt looks for some system-global files to pick up things like the default font, color theme, etc (which makes sense so any Qt apps automatically feel 'native') but my guess is that if the versions are different those files contain entries that the conda-supplied Qt thinks are invalid.

Are you by any chance using KDE as a desktop manager?

**Comment by aweinstein** [comment]
@tacaswell Thank you for your answer. I'm using Cinnamon, not KDE. 

Do you think this is something I should report to Anaconda?

**Comment by tacaswell** [comment]
It looks like cinnamon is Qt based as well so I suspect that is the source of the global style.

You can try, but I am not sure what the resolution would be.  If my theory is correct (a big if!) then anaconda could patch their build of Qt to ignore the global configuration which would fix your problem, but then they would get bug reports that Qt applications no longer respect the global configuration!  

Given that the error is unclear about which setting it is and the issue may be specific to your exact configuration (both Qt/cinnamon settings and all of the Qt versions around), it is very hard for anyone to investigate.  

There may be some Qt specific environmental variables to control what configuration it looks for at runtime.

**Comment by unimamun** [comment]
I have experienced this exact error while running, anaconda-navigator. More badly, my Spyder ide was becoming invisible time to time and I couldn't solve it by resetting spyder from the console. When I downgraded pyqt and qt from version 5.9 to 5.6, both of my problems were solved. Meaning, I neither got the error message and no problem with the Spyder anymore. 

To downgrade run in console:
conda install pyqt=5.6
